### ISSUE TITLE
Translating OBDA to R2RML - Multiple SubjectMaps allowed

### ISSUE THREAD

**Issue by dachafra** [ontop/ontop#461]
### Description

Multiple subject-template rules are allowed for a unique mapping in OBDA. In R2RML one TriplesMap should contain one and only one SubjectMap, but during the translation process with Ontop, multiple SubjectMaps are described in one TriplesMap.

### Steps to Reproduce

1. Download https://github.com/noi-techpark/it.bz.opendatahub.sparql/blob/master/vkg/odh.obda
2. Run `./ontop mapping to-r2rml -i odh.obda -o odh.r2rml`

**Expected behavior:** Each TriplesMap has one and only one SubjectMap

**Actual behavior:** A TriplesMap contain all `rdf:type/a CLASS` rules from the OBDA mapping in multiple SubjectMaps

### Attached material

OBDA Mapping: https://github.com/noi-techpark/it.bz.opendatahub.sparql/blob/master/vkg/odh.obda
R2RML mapping obtained is attached (see, for example, line 171)
[odh.r2rml.txt](https://github.com/ontop/ontop/files/7089027/odh.r2rml.txt)

### Versions

Ontop-Cli 4.1.1

**Comment by kontchakov** [comment]
Hi David

I assume you mean 

 rr:subjectMap [ a rr:SubjectMap, rr:TermMap;
      rr:class schema:GeoCoordinates, geo:Geometry;
      rr:template "http://noi.example.org/data/geo/accommodation/{id}";
      rr:termType rr:IRI
    ], [ a rr:SubjectMap, rr:TermMap;
      rr:template "http://noi.example.org/data/accommodation/{id}";
      rr:termType rr:IRI
    ] .

is not a valid R2RML. Section 6.2 of the R2RML spec (https://www.w3.org/TR/r2rml/#typing), however, says "A subject map may have one or more class IRIs.". Did I miss anything?

Best
Roman

**Comment by dachafra** [comment]
Exactly, this is not a valid SubjectMap. You cannot provide multiple instantiations of SubjectMap in the same TriplesMap.

This part is correct, you are providing one or more class IRIs in the second line
```
rr:subjectMap [ a rr:SubjectMap, rr:TermMap;
rr:class schema:GeoCoordinates, geo:Geometry;
rr:template "http://noi.example.org/data/geo/accommodation/{id}";
rr:termType rr:IRI
],

```

But this is now allowed, you have to create another TriplesMap, with this SubjectMap and the related LogicalTable and POMs
```
 [ a rr:SubjectMap, rr:TermMap;
rr:template "http://noi.example.org/data/accommodation/{id}";
rr:termType rr:IRI
] .
```

**Comment by kontchakov** [comment]
Fixed by 8407c1ab8f3517933694db52635b23ed79bb074b.
